### PR TITLE
Code: Converted `dom` package to TypeScript

### LINKED ISSUE #12182: TypeScript: Convert `dom` package

### PR DESCRIPTION
**Pull request by barklund**
## Context

Trivial TypeScript conversion (all the types are literally strings 😂).

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #12182

### CODE REVIEW

File: packages/dom/src/ensureFontLoaded.ts [hunk 1]
```diff
@@ -19,11 +19,11 @@
  *
  * See https://developer.mozilla.org/en-US/docs/Web/API/FontFaceSet/check
  *
- * @param {string} fontFaceSet A font specification using the CSS value syntax, for example "italic bold 16px Roboto"
- * @param {string} content Limit the font faces to those whose Unicode range contains at least one of the characters in text.
+ * @param fontFaceSet A font specification using the CSS value syntax, for example "italic bold 16px Roboto"
+ * @param content Limit the font faces to those whose Unicode range contains at least one of the characters in text.
  * @return {Promise<boolean>} True if the font is loaded.
```
Comment (miina) [inline packages/dom/src/ensureFontLoaded.ts]:
```suggestion
 * @return True if the font is loaded.
```

**Review by miina — APPROVED** [review]

File: packages/dom/src/stripHTML.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 const buffer = document.createElement('div');
 
-export default function stripHTML(string) {
+export default function stripHTML(string: string) {
   // @todo: implement a cheaper way to strip markup.
   buffer.innerHTML = string;
   return buffer.textContent;
```
Comment (swissspidy) [inline packages/dom/src/stripHTML.ts]: `textContent` can be `null`. Should we force-return a string?

That would be relevant for places like this:

https://github.com/GoogleForCreators/web-stories-wp/blob/4a71cf6ec75cb952bb3868233268ede7550432fb/packages/story-editor/src/components/checklist/utils/characterCountForPage.js#L23-L27

Comment (barklund) [reply packages/dom/src/stripHTML.ts]: Yeah, that makes sense. Fixed in 1256c0c882.

**Review by swissspidy — APPROVED** [review]

**Comment by swissspidy** [comment]
`CUJ: Page Templates: Custom Saved Templates Saved page templates should allow deleting a saved template` test failed, appears flakey.

**Pull request merged**